src/Paket.Core/Versioning/PlatformMatching.fs

```diff
@@ -13,38 +13,45 @@ type ParsedPlatformPath =
     static member Empty = { Name = ""; Platforms = [] }
     static member FromTargetProfile (p:TargetProfile) =
         { Name = p.ToString(); Platforms = p.Frameworks }
-    member x.ToTargetProfile =
+    member x.ToTargetProfile warnIfUnsupported =
         match x.Platforms with
         | _ when System.String.IsNullOrEmpty x.Name -> None
         | [] -> None // Not detected earlier.
         | [p] -> Some (SinglePlatform p)
-        | plats -> Some (TargetProfile.FindPortable plats)
+        | plats -> Some (TargetProfile.FindPortable warnIfUnsupported plats)
     member pp.IsEmpty = String.IsNullOrEmpty pp.Name || pp.Platforms.IsEmpty
 
 let inline split (path : string) =
     path.Split('+')
-    |> Array.map (fun s -> System.Text.RegularExpressions.Regex.Replace(s, @"portable[\d\.]*-",""))
+    |> Array.map (fun s -> System.Text.RegularExpressions.Regex.Replace(KnownAliases.normalizeFramework s, @"portable[\d\.]*-",""))
 
 // TODO: This function does now quite a lot, there probably should be several functions.
-let extractPlatforms = memoize (fun path ->
+let private extractPlatformsPriv = memoize (fun path ->
     if System.String.IsNullOrEmpty path then Some ParsedPlatformPath.Empty
     else
         let splits = split path
         let platforms = splits |> Array.choose FrameworkDetection.Extract |> Array.toList
         if platforms.Length = 0 then
-            if splits.Length = 1 && splits.[0].ToLowerInvariant().StartsWith "profile" then
+            if splits.Length = 1 && splits.[0].StartsWith "profile" then
                 // might be something like portable4.6-profile151
                 let found =
                     KnownTargetProfiles.FindPortableProfile splits.[0]
                     |> ParsedPlatformPath.FromTargetProfile
                 Some { found with Name = path }
             else
-                traceWarnfn "Could not detect any platforms from '%s'" path
                 None
         else Some { Name = path; Platforms = platforms })
 
+let extractPlatforms warn path =
```

Comment:
The general guideline for warning parameter is:
 - true is always OK
 - false is only OK when either the calling method prints a better warning OR when the calling method has it's own warning parameter

This way warnings should not get lost. I don't know if there is better way to do this.